### ISSUE TITLE
Feature Request: Search

### ISSUE THREAD

**Issue by cloudybyte** [charmbracelet/soft-serve#137]
Hey there,
first of all I want to say that I'm really impressed by you've done so far and I really want this to become production ready one day.
However I got a question for you: Given that I would want to migrate my repos over to soft-serve, is there a way to search them?

**Comment by aymanbagabas** [comment]
> Hey there, first of all I want to say that I'm really impressed by you've done so far and I really want this to become production ready one day.

Hi @cloudybyte, I'm glad that you like Soft Serve.

> However I got a question for you: Given that I would want to migrate my repos over to soft-serve, is there a way to search them?

Could you be more specific here? Where are you migrating your repositories from? Do you mean searching the list of repositories hosted in a Soft Serve server?

**Comment by cloudybyte** [comment]
yes, exactly that. I'd probably migrate from GitHub private repos, but doesn't really matter since I got local copies of them

**Comment by aymanbagabas** [comment]
You can press <kbd>/</kbd> to filter repositories in the list  
![image](https://user-images.githubusercontent.com/3187948/183673278-0fdd23c6-0056-4d27-a74d-752c0442d64d.png)

**Comment by cloudybyte** [comment]
Oh ok, I didn't know that... Is there some sort of cheat sheet or wiki?

**Comment by cloudybyte** [comment]
Or is there even deep search? (being able to search for functions and symbols in code)

**Comment by cloudybyte** [comment]
The first thing I can think of is trying out [meilisearch](https://github.com/meilisearch/meilisearch), though that would result in another binary to run

**Comment by everdrone** [comment]
[Typesense](https://github.com/typesense/typesense) is also very powerful but still requires another binary running

**Comment by cloudybyte** [comment]
That would be an option though I don't know how to evaluate what would suit best in this context... Do you have any experience with any of those @everdrone?

**Comment by everdrone** [comment]
@cloudybyte I've used Typesense in the past, it would require a separate installation plus a systemd service file (or separate docker container).

Then we could add the client token to `config.yml` for `soft` to search.

Each time a new repo is detected we have to index it into the repo collection, and delete it when not detected by `soft` anymore.

Plus we could index some other information like how many times it has been cloned through soft, but I'm not sure how that could work specifically

**Comment by cloudybyte** [comment]
Afaik that would be pretty much the same process meilisearch would require as well, though I haven't used it yet.
